bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutationGCJClient.java

```diff
@@ -17,54 +17,48 @@
 
 import com.google.api.core.ApiFuture;
 import com.google.api.core.InternalApi;
-import com.google.cloud.bigtable.config.Logger;
+import com.google.api.gax.batching.Batcher;
 import com.google.cloud.bigtable.core.IBulkMutation;
-import com.google.cloud.bigtable.data.v2.models.BulkMutationBatcher;
-import com.google.cloud.bigtable.data.v2.models.BulkMutationBatcher.BulkMutationFailure;
-import com.google.cloud.bigtable.data.v2.models.RowMutation;
-import com.google.cloud.bigtable.util.ApiFutureUtil;
+import com.google.cloud.bigtable.data.v2.models.RowMutationEntry;
 import com.google.common.base.Preconditions;
 import java.io.IOException;
-import java.util.concurrent.TimeoutException;
 
 /**
  * This class is meant to replicate existing {@link BulkMutation} while translating calls to
- * Google-Cloud-Java's {@link BulkMutationBatcher} api.
+ * Google-Cloud-Java's {@link Batcher} api.
  */
 @InternalApi("For internal usage only")
 public class BulkMutationGCJClient implements IBulkMutation {
 
-  private static Logger LOG = new Logger(BulkMutationGCJClient.class);
+  private final Batcher<RowMutationEntry, Void> bulkMutateBatcher;
 
-  private final BulkMutationBatcher bulkMutateBatcher;
-  private final OperationAccountant operationAccountant;
-
-  public BulkMutationGCJClient(BulkMutationBatcher bulkMutateBatcher) {
+  public BulkMutationGCJClient(Batcher<RowMutationEntry, Void> bulkMutateBatcher) {
     this.bulkMutateBatcher = bulkMutateBatcher;
-    this.operationAccountant = new OperationAccountant();
   }
 
   /** {@inheritDoc} */
   @Override
-  public synchronized ApiFuture<Void> add(RowMutation rowMutation) {
+  public synchronized ApiFuture<Void> add(RowMutationEntry rowMutation) {
     Preconditions.checkNotNull(rowMutation, "mutation details cannot be null");
-    final ApiFuture<Void> response = bulkMutateBatcher.add(rowMutation);
-    operationAccountant.registerOperation(ApiFutureUtil.adapt(response));
-    return response;
+    return bulkMutateBatcher.add(rowMutation);
   }
 
   /** {@inheritDoc} */
   @Override
   public void sendUnsent() {
-    LOG.info("This operation will be implemented once the underlying API has this feature.");
+    try {
+      bulkMutateBatcher.flush();
```

Comment:
shouldnt this be non-blocking?

Reply:
Do you think I should execute it in the separate thread something like:
```java
  /** {@inheritDoc} */
  @Override
  public void sendUnsent() {
    new Thread() {
      @Override
      public void run() {
        try {

          bulkMutateBatcher.flush();
        } catch (InterruptedException ex) {
          throw new RuntimeException("Could not complete RPC for current Batch", ex);
        }
      }
    }.start();
  }

```

Reply:
I just meant you should use sendOutstanding() instead of flush. Unless I'm mistaken this method is used for async operations in hbase that return a future and we just needed something start the request

Reply:
ping